clang/lib/Sema/SemaExpr.cpp

```diff
@@ -16538,6 +16538,13 @@ ExprResult Sema::BuildVAArgExpr(SourceLocation BuiltinLoc,
         << TInfo->getTypeLoc().getSourceRange();
     }
 
+    if (TInfo->getType()->isVariableArrayType()) {
```

Comment:
This should check for any array type; the issue with compatibility isn't specific to variadic arrays.